git.go

```diff
@@ -40,79 +72,40 @@ func openOrCloneRepo() error {
 }
 
 func syncRepo(repo *git.Repository) error {
-	remote, err := repo.Remotes.Lookup("origin")
+	wt, err := repo.Worktree()
 	if err != nil {
-		return errors.New("Couldn't lookup remote: " + err.Error())
+		return errors.New("Couldn't get WorkTree: " + err.Error())
 	}
-	if err := remote.Fetch([]string{}, &git.FetchOptions{
-		RemoteCallbacks: git.RemoteCallbacks{
-			CredentialsCallback:      credentialsCallback,
-			CertificateCheckCallback: certificateCheckCallback,
-		},
-	}, ""); err != nil {
-		return errors.New("Couldn't fetch remote: " + err.Error())
-	}
-	// Merge
-	remoteRef, err := repo.References.Lookup("refs/remotes/origin/" + viper.GetString("repo.branch"))
-	if err != nil {
-		return errors.New("Couldn't lookup ref: " + err.Error())
-	}
-	mergeRemoteHead, err := repo.AnnotatedCommitFromRef(remoteRef)
-	if err != nil {
-		return errors.New("Couldn't get commit ref: " + err.Error())
-	}
-
-	mergeHeads := make([]*git.AnnotatedCommit, 1)
-	mergeHeads[0] = mergeRemoteHead
-	if err = repo.Merge(mergeHeads, nil, nil); err != nil {
-		return errors.New("Couldn't merge: " + err.Error())
-	}
-	if err := repo.CheckoutHead(nil); err != nil {
-		return errors.New("Couldn't checkout head: " + err.Error())
+	if err := wt.Pull(&git.PullOptions{}); err != nil && err != git.NoErrAlreadyUpToDate {
+		return errors.New("Couldn't pull: " + err.Error())
 	}
 	head, err := repo.Head()
 	if err != nil {
 		return errors.New("Couldn't checkout head: " + err.Error())
 	}
-	commit, err := repo.LookupCommit(head.Target())
+	commit, err := repo.CommitObject(head.Hash())
 	if err != nil {
-		return errors.New("Couldn't lookup commit: " + err.Error())
+		return errors.New("Couldn't get commit: " + err.Error())
 	}
 	tree, err := commit.Tree()
 	if err != nil {
 		return errors.New("Couldn't get commit tree: " + err.Error())
 	}
-	tree.Walk(walkCallback)
-	log.Info("Repo synced")
-	return nil
-}
```

Comment:
In the `syncRepo` routine it seems that when using `ssh` based cloning it will then try to connect to an `ssh-agent` for any following `sync` (here a `git pull`) which will error in most production environment (for example we do not run a `ssh-agent` in a container). It seems weird to me that a clone would work but not a pull. This should be investigated before this PR is merged.

- Returned error on a clean environment (no `ssh-agent` or `gpg-agent` with ssh support started)

```
WARN[0000] Couldn't find repo locally, trying to clone it  error="repository does not exist"
INFO[0000] Cloning repo                                  branch=my-branch path=data/ url="git@github.com:0rax/my-private-repo.git"
WARN[0001] Couldn't sync repo                            error="Couldn't pull: error creating SSH agent: \"SSH agent requested but SSH_AUTH_SOCK not-specified\""```